File: CompareProgrammingLanguage/webname-csharp/webname/conversion/web2csv.cs
using System.Text;

namespace WebName
{
    public class WriteFile {
        public void web2csv(WebNameTable web) {
            StringBuilder csv_string = new StringBuilder();
            csv_string.Append(web.column_name[0]);

            for (int x = 1; x < web.column_name.Count; x++) {
                 csv_string.Append(",");
                 csv_string.Append(web.column_name[x]);
            }

            csv_string.Append("\n");

            for (int y = 1; y < web.fact_table[0].Count; y++) {
                for (int x = 0; x < web.column_name.Count; x++) {
                    if(x > 0) {
                       csv_string.Append(",");
                    }

                    if(web.data_type[x] != "Number") {                       
                        csv_string.Append(web.key2value[x][Convert.ToInt64(web.fact_table[x][y])]);
                    }
                    else {
                        csv_string.Append(web.fact_table[x][y].ToString());
                    }                   
                }
                csv_string.Append("\n");
            }

            using (StreamWriter toDisk = new StreamWriter("data.csv"))
            {
                toDisk.Write(csv_string);
                toDisk.Close();
            }
        }
    }


}

File: CompareProgrammingLanguage/webname-rust/src/webname/conversion/web2csv.rs
use std::io::Read;
use std::io::Write;
use std::fs::File;
use std::fs;

pub mod csv2web;

pub fn web2csv(web: csv2web::WebName) {
    let mut csv_string = String::new();
    csv_string.push_str(&web.column_name[0]);

    for x in 1..web.column_name.len() {         
          csv_string.push_str(",");
          csv_string.push_str(&web.column_name[x]);
    }

    csv_string.push_str("\n");

    for y in 1..web.fact_table[&0].len() {                
        for x in 0..web.column_name.len() {              
            if x > 0 {
               csv_string.push_str(",");               
            }

            if web.data_type[x] != "Number" {            
                csv_string.push_str(&web.key2value[&x][&(web.fact_table[&x][y] as usize)]);     
            }       
            else {                               
               csv_string.push_str(&web.fact_table[&x][y].to_string())
            }
        }
        csv_string.push_str("\n");
    }

    let mut file = std::fs::File::create("data.csv").expect("create failed");
    file.write_all(csv_string.as_bytes()).expect("write failed");
}







